src/modules/tab_completion/index.js

```diff
@@ -0,0 +1,109 @@
+const $ = require('jquery');
+// const debug = require('../../utils/debug');
+const keyCodes = require('../../utils/keycodes');
+const emotes = require('../emotes');
+// const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+const CHAT_TEXT_AREA = '.ember-chat .chat-interface textarea';
+const CONVERSATION_TEXT_AREA = '.conversation-window .chat-input textarea';
+const CHAT_SUGGESTIONS = '.ember-chat .chat-interface .suggestions';
+
+class TabCompletionModule {
+    /*
+    TODO:
+        - priority ordering
+    */
+    constructor() {
+        this.load();
+        watcher.on('chat.message', ($el, msg) => this.storeUser($el, msg));
+        watcher.on('load.chat', () => this.resetChannelData());
+    }
+
+    load() {
+        this.tabTries = -1;
+        this.suggestions = null;
+        this.textSplit = ['', '', ''];
+        this.userList = new Set();
+
+        // Chat
+        $('body').off('click focus', CHAT_TEXT_AREA).on('click focus', CHAT_TEXT_AREA, this.onFocus);
+        $('body').off('keydown', CHAT_TEXT_AREA).on('keydown', CHAT_TEXT_AREA, e => this.onKeydown(e));
```

Comment:
it's safe to `off` this without the callback because we're using our own jquery, but we shouldn't do this as it means no other module can listen to his event on that textarea